api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -219,6 +220,12 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 						return outputs, false, v1alpha2.NewCOAError(nil, fmt.Sprintf("Invalid solution name: catalog - %s", name), v1alpha2.BadRequest)
 					}
 
+					if annotation_name != "" {
```

Comment:
You can add a function in objectMetadata to add the annotation, avoid writing the same code here.